### PR TITLE
Readme: fix gitter link, add Nim logo, formatting fixes

### LINKED ISSUE #39: The gitter link in README.md is not working.

### PR DESCRIPTION
**Pull request by ZoomRmc**
* Fixes the gitter badge link
* Added Matrix badge and links
* Use more extended markdown syntax instead of HTML in header

This PR supersedes PR #40 by @dlesnoff, sorry for this. 
I tried to suggest a quick fix there and broke the formatting, and couldn't suggest additional fixes that include the starting lines that were not touched by the original commit there, so an alternative PR seemed like the easiest way to quickly fix everything.
Fixes #39.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,40 +1,34 @@
...
-
-We are on [Discord](https://the-algorithms.com/discord) and [Gitter](https://gitter.im/TheAlgorithms)! Community channels are a great way for you to ask questions and get help. Please join us!
-
 ## List of Algorithms
 
 See our [directory](DIRECTORY.md) for easier navigation and a better overview of the project.
+
+## Community Channels
+
+We are on [Discord](https://the-algorithms.com/discord) and [Gitter](https://gitter.im/TheAlgorithms)/[Matrix](https://matrix.to/#/#TheAlgorithms_community:gitter.im)!
```
Comment (dlesnoff) [inline README.md:25]:
```suggestion
We are on [Discord](https://the-algorithms.com/discord) and [Gitter](https://gitpod.io/#https://github.com/TheAlgorithms/Nim)/[Matrix](https://matrix.to/#/#TheAlgorithms_community:gitter.im)!
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# <a href="https://github.com/TheAlgorithms/">The Algorithms</a> – Nim&nbsp;<img src="https://raw.githubusercontent.com/nim-lang/assets/master/Art/logo-crown.png" height="28px"/><img align="right" src="https://raw.githubusercontent.com/TheAlgorithms/website/main/public/logo.svg" height="100">
+# <a href="https://github.com/TheAlgorithms/">The Algorithms</a> – Nim&nbsp;<img src="https://raw.githubusercontent.com/nim-lang/assets/master/Art/logo-crown.png" height="28px"/><a href="https://github.com/TheAlgorithms/"><img align="right" src="https://raw.githubusercontent.com/TheAlgorithms/website/main/public/logo.svg" height="100"></a>
```
Comment (dlesnoff) [inline README.md:1]: The image already links to the website, not the Github page. Why do you add a link here?

Comment (ZoomRmc) [reply README.md:1]: Before this change it linked to the image file itself, not like it does at the current state of the Readme.

**Comment by dlesnoff** [comment]
All the informations are much more compacted. I liked that the Python README.md was the very first md file that I encountered that was not left-aligned due to this MD limitation. It doesn't look bad though!

We should have a link to the website under the list of algorithms section.

File: README.md [hunk 3]
```diff
@@ -1,40 +1,38 @@
...
 Implementations are for learning purposes only. They may be less efficient than the implementations in the [Nim standard library](https://nim-lang.org/docs/lib.html) or [Nimble packages](https://nimble.directory/). Use them at your discretion.
 
 ## Getting Started
 
 Read through our [Contribution Guidelines](CONTRIBUTING.md) before you contribute.
 
+## List of Algorithms
+
+See our [directory](DIRECTORY.md) for easier navigation and a better overview of the project.
+
```
Comment (dlesnoff) [inline README.md:22]:
```suggestion

You can also browse the implementations on [the website](https://the-algorithms.com/language/nim)
```

Comment (dlesnoff) [reply README.md:22]: Oh no I edited it too late. Add a point!
```suggestion


You can also browse the implementations on [the website](https://the-algorithms.com/language/nim).
```

**Review by dlesnoff — APPROVED** [review]
Nice style improvement and fixes!

**Comment by ZoomRmc** [comment]
That's a hilarious mess! The easier I think it is, the more silly mistakes I made!
Thanks for the fixes.

**Comment by dlesnoff** [comment]
Can I ask you to do another PR to fix the CI?

**Pull request merged**